File: MATLAB/writeFloatingDouble.m
function writeFloatingDouble(signal, filename)
    % Write floating-point (double) values to a text file, one per line.
    % Keeps full double precision with %.17g formatting.

    fid = fopen(filename, 'w');
    if fid == -1
        error('Cannot open file %s for writing', filename);
    end

    % Convert to double if needed
    signal = double(signal);

    % Write each sample with full precision (round-trip safe)
    for i = 1:length(signal)
        fprintf(fid, '%.17g\n', signal(i));
    end

    fclose(fid);
end

File: Python_Script/write_floating_double.py
import os

def write_floating_double(signal_data, filename):
    """
        Write floating-point (double) values to a text file, one per line.
    """

    os.makedirs(os.path.dirname(filename), exist_ok = True)

    with open(filename, 'w') as f:
        for sample in signal_data:
            f.write(f"{sample:.17g}\n")